src/processing/SpatialIndex.cpp

```diff
@@ -127,38 +132,45 @@ void SpatialIndex::processChangedCS(entity::SpatialReference::Ptr cs)
 
 SpatialIndex::bValue SpatialIndex::createEntry(entity::Geometry::Ptr geo)
 {
+    
     // get the 3d envelope of the geometry.
-    OGREnvelope3D envelope;
-    geo->geometry()->getEnvelope(&envelope);
+    geos::geom::Coordinate min, max;
+    geo->findEnvelope(min, max);
+
     // this envelope is in the coordinate system of the geometry. But what we need is an envelope
     // that is axis aligned with the root reference system. We could transform the geometry to root,
     // take the envelope and transform it back, but that is just ridiculus. Instead: Create a
     // bounding-box-geometry (8 points, one for each corner), transform it, and take its envelope.
     // ---
     // create a geometry with the matching extends: 8 point, every corner must be checked!
-    OGRMultiPoint mp;
-    OGRPoint* p;
-    p = new OGRPoint(envelope.MinX, envelope.MinY, envelope.MinZ); mp.addGeometryDirectly(p);
-    p = new OGRPoint(envelope.MinX, envelope.MinY, envelope.MaxZ); mp.addGeometryDirectly(p);
-    p = new OGRPoint(envelope.MinX, envelope.MaxY, envelope.MinZ); mp.addGeometryDirectly(p);
-    p = new OGRPoint(envelope.MinX, envelope.MaxY, envelope.MaxZ); mp.addGeometryDirectly(p);
-    p = new OGRPoint(envelope.MaxX, envelope.MinY, envelope.MinZ); mp.addGeometryDirectly(p);
-    p = new OGRPoint(envelope.MaxX, envelope.MinY, envelope.MaxZ); mp.addGeometryDirectly(p);
-    p = new OGRPoint(envelope.MaxX, envelope.MaxY, envelope.MinZ); mp.addGeometryDirectly(p);
-    p = new OGRPoint(envelope.MaxX, envelope.MaxY, envelope.MaxZ); mp.addGeometryDirectly(p);
+
+    geos::geom::Coordinate coord;
+    std::vector<geos::geom::Coordinate> cornerCoordinates;
+    coord = geos::geom::Coordinate(min.x, min.y, min.z); cornerCoordinates.push_back(coord);
+    coord = geos::geom::Coordinate(min.x, min.y, max.z); cornerCoordinates.push_back(coord);
+    coord = geos::geom::Coordinate(min.x, max.y, min.z); cornerCoordinates.push_back(coord);
+    coord = geos::geom::Coordinate(min.x, max.y, max.z); cornerCoordinates.push_back(coord);
+    coord = geos::geom::Coordinate(max.x, min.y, min.z); cornerCoordinates.push_back(coord);
+    coord = geos::geom::Coordinate(max.x, min.y, max.z); cornerCoordinates.push_back(coord);
+    coord = geos::geom::Coordinate(max.x, max.y, min.z); cornerCoordinates.push_back(coord);
+    coord = geos::geom::Coordinate(max.x, max.y, max.z); cornerCoordinates.push_back(coord);
+
+    // todo will this be cleaned up correctly ?
+    geos::geom::MultiPoint* mp = geos::geom::GeometryFactory::getDefaultInstance()->createMultiPoint(cornerCoordinates);
```

Comment:
No, it won't be cleaned up correctly. In the old version with GDAL I used a "OGRMultiPoint" here, which was created on the stack, not through the factory. Since you use the factory here to create the object you'll have to delete it, too.

Reply:
Solved in 2de0f2f0c989cab545f93c227bf8983f3bdc5f98